tools/apidiff/Makefile

```diff
@@ -380,36 +387,40 @@ $(APIDIFF_DIR)/updated-references/xm/%.xml: $(MAC_DESTDIR)$(MAC_FRAMEWORK_CURREN
 
 # The dotnet references xmls may come from different hashes, so we need to have separate rules for all of them
 $(APIDIFF_DIR)/references/dotnet/legacy-diff/Microsoft.iOS.Ref/ref/net6.0/Xamarin.iOS.xml: $(UNZIP_DIR_DOTNET_iOS)/Microsoft.iOS.Ref/ref/net6.0/Xamarin.iOS.dll  $(MONO_API_INFO)
-	$(Q) mkdir -p $(dir $@)
+	$(Q) mkdir -p $(dir $@) $(dir $(APIDIFF_DIR)/references/dotnet/Microsoft.iOS.Ref/ref/net6.0/Xamarin.iOS)
```

Comment:
This looks strange, why would you need to create this additional directory when this rule doesn't write to that directory?

Reply:
@rolfbjarne Okay updated to remove these legacy-diff references! Works locally, should be good on the CI too.